### DISCUSSION TITLE
WebSocket is closed before the connection is established.

### DISCUSSION THREAD

**Discussion by KristianJohansenVakwerk** [sanity-io/sanity#6452] [Sanity Studio Support]
Hey! 

I have tried to get the presentation tool implmemented into my v3 embedded studio without luck. I keep getting this warning: "WebSocket is closed before the connection is established." I have followed the guide on your website and use the repo below as an example.  I hope someone can help. Thanks!

https://github.com/sanity-io/template-nextjs-personal-website/tree/main

![Group 1](https://github.com/sanity-io/sanity/assets/46603753/a016a64a-3c6b-4875-9ca0-4cc5882b7c67)

**Comment by LiFaytheGoblin** [comment]
Maybe related to this: I get the errors `Firefox can’t establish a connection to the server at wss://<ID CENSORED>.api.sanity.io/v2022-06-30/socket/production?tag=sanity.studio.` and `The connection to wss://<ID CENSORED>.api.sanity.io/v2022-06-30/socket/production?tag=sanity.studio was interrupted while the page was loading.` with Sanity 3.48.1

**Reply by LiFaytheGoblin** [reply]
I have ignored this issue ever since (the errors are still logged to the console) and the studio and data are fine

**Comment by considwilliam** [comment]
We are getting the same error when trying to implement visual editing. Did you end up solving it? @KristianJohansenVakwerk and @LiFaytheGoblin?

**Reply by maria-program** [reply]
Did you solve it? @considwilliam I am now having the same issue

**Comment by tarjep** [comment]
I am also having issues with this. Any solutions?

**Comment by gitknul** [comment]
I am also running into this problem, using `3.61.0`

**Comment by zoewaldman** [comment]
no one solved this?

**Comment by gitknul** [comment]
I was able to fix by using the correct config; apparently the config file depended on some server only code. I removed that dependency from that file and Studio managed to load. Refer to a worknig example below:

`src/app/studio/[[...tool]]/page.tsx`
```
/**
 * This route is responsible for the built-in authoring environment using Sanity Studio.
 * All routes under your studio path is handled by this file using Next.js' catch-all routes:
 * https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes
 *
 * You can learn more about the next-sanity package here:
 * https://github.com/sanity-io/next-sanity
 */

import { NextStudio } from 'next-sanity/studio';
import config from '../../../../sanity.config';

export const dynamic = 'force-static';

export { metadata, viewport } from 'next-sanity/studio';

export default async function StudioPage() {
  return <NextStudio config={config} />;
}
```

`sanity.config.ts`
```
'use client'; // <-- IMPORTANT - add this directive

/**
 * This configuration is used to for the Sanity Studio that’s mounted on the `/app/studio/[[...tool]]/page.tsx` route
 */
import { visionTool } from '@sanity/vision';
import { defineConfig, DocumentActionComponent } from 'sanity';
import { structureTool } from 'sanity/structure';

import { schemaTypes, singletonTypeNames } from '@/sanity/schemaTypes';
import { structure } from '@/sanity/structure';
import { defaultDocumentNode } from '@/sanity/structure/defaultDocumentType';

const studioConfig = {
  apiVersion: 'v2022-03-07', // don't set through env, as migrating to another version could involve breaking changes
  dataset: assertValue(process.env.NEXT_PUBLIC_SANITY_STUDIO_DATASET, 'Missing environment variable: NEXT_PUBLIC_SANITY_STUDIO_DATASET'),
  projectId: assertValue(process.env.NEXT_PUBLIC_SANITY_STUDIO_PROJECT_ID, 'Missing environment variable: NEXT_PUBLIC_SANITY_STUDIO_PROJECT_ID'),
};

function assertValue(value: string | undefined, errorMessage: string): string {
  if (value === undefined || !value) {
    throw new Error(errorMessage);
  }

  return value;
}

const { projectId, dataset, apiVersion } = studioConfig;

export default defineConfig({
  name: 'default',
  title: `gitknul`,
  basePath: '/studio',
  projectId,
  dataset,
  plugins: [structureTool({ structure, defaultDocumentNode }), visionTool({ defaultApiVersion: apiVersion })],

  schema: {
    types: schemaTypes,
  },
});

```

**Reply by Race428** [reply]
Is it an issue that the API version is 2022?

**Reply by gitknul** [reply]
I always use that API version without encountering any problems.

**Reply by zoewaldman** [reply]
Hey @gitknul I still haven't been able to resolve this. I just spent the day trying to isolate my embedded studio from my project because it was fetching my sites header / footer. That's resolved now but still getting the websocket error. 

How do you ensure / know what API version to use? Any other tips on resolving the websocket error? Is it safe to ignore if I'm still able to access the studio?

**Comment by cheryl-c-tirerack** [comment]
anyone?

**Comment by adith-sarvam** [comment]
Still facing the same issue. But the studio seems to work fine. Can anyone confirm if its safe to ignore?

**Comment by hakimmazouz** [comment]
I'm dealing with this too.
